R/animint.R

```diff
@@ -1285,6 +1350,46 @@ animint2dir <- function(plot.list, out.dir = tempfile(),
     }
   }
 
+  ## Selector levels and update were stored in saveLayer, so now
+  ## compute the unique values to store in meta$selectors.
+  for(selector.name in names(meta$selector.values)){
+    values.update <- meta$selector.values[[selector.name]]
+    value.vec <- unique(unlist(lapply(values.update, "[[", "values")))
+    meta$selectors[[selector.name]]$selected <- if(
+      meta$selectors[[selector.name]]$type=="single"){
+      value.vec[1]
+    }else{
+      value.vec
+    }
+    ## If this selector was defined by .variable .value aes, then we
+    ## will not generate selectize widgets. This is indicated by the
+    ## compiler by not setting the "levels" attribute of the selector.
+    if(!isTRUE(meta$selectors[[selector.name]]$is.variable.value)){
+      meta$selectors[[selector.name]]$levels <- value.vec
+    }
+    ## s.info$update is the list of geom names that will be updated
+    ## for this selector.
+    meta$selectors[[selector.name]]$update <-
+      as.list(unique(unlist(lapply(values.update, "[[", "update"))))
+  }
+
+  ## For a static data viz with no interactive aes, no need to check
+  ## for trivial showSelected variables with only 1 level.
+  if(0 < length(meta$selectors)){
+    n.levels <- sapply(meta$selectors, function(s.info)length(s.info$levels))
+    one.level <- n.levels == 1
+    has.legend <- sapply(meta$selectors, function(s.info)isTRUE(s.info$legend))
+    is.trivial <- one.level & (!has.legend)
+    if(any(is.trivial)){
+      ## With the current compiler that has already saved the tsv files
+      ## by now, we can't really make this data viz more efficient by
+      ## ignoring this trivial selector. However we can warn the user so
+      ## that they can remove this inefficient showSelected.
+      warning("showSelected variables with only 1 level: ",
+              paste(names(meta$selectors)[is.trivial], collapse=", "))
+    }
```

Comment:
Hey @kferris10 thanks for the idea about the possibility of a showSelected variable with only 1 level. In that case we warn the user if it did not come from a legend with 1 entry (which is also a pretty weird situation, but possible, as in `aes(fill=place)` in https://github.com/tdhock/animint/blob/master/tests/testthat/test-renderer2-interactivity.R)